src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -2840,13 +2821,13 @@ emit_vector_2_3_4 (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *f
 		if (!(fsig->param_count == 2 && mono_metadata_type_equal (fsig->params [0], type) && mono_metadata_type_equal (fsig->params [1], type)))
 			return NULL;
 		MonoClass *arg_class = mono_class_from_mono_type_internal (fsig->params [0]);
-		return emit_xequal (cfg, arg_class, args [0], args [1]);
+		return emit_xequal (cfg, arg_class, MONO_TYPE_R4, args [0], args [1]);
```

Comment:
Why hard-coding the type?

Reply:
It is hard-coded for a lot of emit_vector_2_3_4 methods, those can only be of type Single.

Reply:
Let me re-frame my question. Is `get_underlying_type (m_class_get_this_arg (arg1->klass))` not returning `MONO_TYPE_R4` for `Vector4`?

Reply:
Exactly, for `Vector4` that line translates to `get_underlying_type (MonoType* type)` which returns `type->type` , i.e `MONO_TYPE_VALUETYPE`.

Reply:
I see. Then I am okay with passing the type like this.